satpy/tests/reader_tests/test_eum_l2_grib.py

```diff
@@ -0,0 +1,299 @@
+    "Ny": 5568,
+    "earthMajorAxis": 6378140.,
+    "earthMinorAxis": 6356755.,
+    "NrInRadiusOfEarth": 6.6107,
+    "XpInGridLengths": 2784.0,
+    "parameterNumber": 30,
+    "missingValue": 9999,
+}
+
+# List to be used as fake GID source
+FAKE_GID = [0, 1, 2, 3, None]
+
+
+class TestEUML2GribFileHandler(unittest.TestCase):
+    """Test the EUML2GribFileHandler reader."""
+
+    @mock.patch("satpy.readers.eum_l2_grib.ec")
+    def setUp(self, ec_):
+        """Set up the test by creating a mocked eccodes library."""
+        fake_gid_generator = (i for i in FAKE_GID)
+        ec_.codes_grib_new_from_file.side_effect = lambda fh: next(fake_gid_generator)
+        self.ec_ = ec_
+
+    def common_checks(self, mock_file, dataset_id):
+        """Commmon checks for fci and seviri data."""
+        # Checks that the codes_grib_multi_support_on function has been called
+        self.ec_.codes_grib_multi_support_on.assert_called()
+
+        # Restarts the id generator and clears the call history
+        fake_gid_generator = (i for i in FAKE_GID)
+        self.ec_.codes_grib_new_from_file.side_effect = lambda fh: next(fake_gid_generator)
+        self.ec_.codes_grib_new_from_file.reset_mock()
+        self.ec_.codes_release.reset_mock()
+
+        # Checks the correct execution of the get_dataset function with a valid parameter_number
+        valid_dataset = self.reader.get_dataset(dataset_id, {"parameter_number": 30})
+        # Checks the correct file open call
+        mock_file.assert_called_with("test.grib", "rb")
+        # Checks that the dataset has been created as a DataArray object
+        assert valid_dataset._extract_mock_name() == "xr.DataArray()"
+        # Checks that codes_release has been called after each codes_grib_new_from_file call
+        # (except after the last one which has returned a None)
+        assert self.ec_.codes_grib_new_from_file.call_count == self.ec_.codes_release.call_count + 1
+
+        # Restarts the id generator and clears the call history
+        fake_gid_generator = (i for i in FAKE_GID)
+        self.ec_.codes_grib_new_from_file.side_effect = lambda fh: next(fake_gid_generator)
+        self.ec_.codes_grib_new_from_file.reset_mock()
+        self.ec_.codes_release.reset_mock()
+
+        # Checks the correct execution of the get_dataset function with an invalid parameter_number
+        invalid_dataset = self.reader.get_dataset(dataset_id, {"parameter_number": 50})
+        # Checks that the function returns None
+        assert invalid_dataset is None
+        # Checks that codes_release has been called after each codes_grib_new_from_file call
+        # (except after the last one which has returned a None)
+        assert self.ec_.codes_grib_new_from_file.call_count == self.ec_.codes_release.call_count + 1
+
+    @unittest.skipIf(sys.platform.startswith("win"), "'eccodes' not supported on Windows")
+    @mock.patch("satpy.readers.eum_l2_grib.xr")
+    @mock.patch("satpy.readers.eum_l2_grib.da")
```

Comment:
I see a bit of mocking. In our experience this can lead to problems down the road, and nowadays, we prefer to write a synthetic testing file to tmp_path and work with that instead. Do you think this would be doable for this format?

Reply:
To be honest I don't know and I also don't have the time to look into it at this point. Would be grateful if we could tackle that in another PR in the future :)

Reply:
Yes, would be a pity if we were to forget it 😃  I created an issue here: https://github.com/pytroll/satpy/issues/2932